backend/pkg/console/schema_registry.go

```diff
@@ -0,0 +1,569 @@
+// soft-deleted subjects.
+func (s *Service) GetSchemaRegistrySubjects(ctx context.Context) ([]SchemaRegistrySubject, error) {
+	subjects := make(map[string]struct{})
+	subjectsWithDeleted := make(map[string]struct{})
+
+	grp, _ := errgroup.WithContext(ctx)
+	grp.Go(func() error {
+		res, err := s.kafkaSvc.SchemaService.GetSubjects(ctx, false)
+		if err != nil {
+			return err
+		}
+		for _, subject := range res.Subjects {
+			subjects[subject] = struct{}{}
+		}
+		return nil
+	})
+	grp.Go(func() error {
+		res, err := s.kafkaSvc.SchemaService.GetSubjects(ctx, true)
+		if err != nil {
+			return err
+		}
+		for _, subject := range res.Subjects {
+			subjectsWithDeleted[subject] = struct{}{}
+		}
+		return nil
+	})
+	if err := grp.Wait(); err != nil {
+		return nil, err
+	}
+
+	result := make([]SchemaRegistrySubject, 0, len(subjectsWithDeleted))
+	for subj := range subjectsWithDeleted {
+		_, exists := subjects[subj]
+		result = append(result, SchemaRegistrySubject{
+			Name:          subj,
+			IsSoftDeleted: !exists,
+		})
+	}
+
+	// Sort for stable results in UI
+	slices.SortFunc(result, func(a, b SchemaRegistrySubject) int {
+		return strings.Compare(a.Name, b.Name)
+	})
+
+	return result, nil
+}
+
+// SchemaRegistrySubjectDetails represents a schema registry subject along
+// with other information such as the registered versions that belong to it,
+// or the full schema information that's part of the subject.
+type SchemaRegistrySubjectDetails struct {
+	Name                string                                `json:"name"`
+	Type                schema.SchemaType                     `json:"type"`
+	Compatibility       schema.CompatibilityLevel             `json:"compatibility"`
+	RegisteredVersions  []SchemaRegistrySubjectDetailsVersion `json:"versions"`
+	LatestActiveVersion int                                   `json:"latestActiveVersion"`
+	Schemas             []*SchemaRegistryVersionedSchema      `json:"schemas"`
+}
+
+const (
+	// SchemaVersionsAll can be specified as version to retrieve all schema versions.
+	SchemaVersionsAll string = "all"
+	// SchemaVersionsLatest can be specified as version to retrieve the latest active schema.
+	SchemaVersionsLatest string = "latest"
+)
+
+// GetSchemaRegistrySubjectDetails retrieves the schema details for the given subject, version tuple.
+// Use version = 'latest' to retrieve the latest schema.
+// Use version = 'all' to retrieve all schemas for this subject.
+// Use version = 3 to retrieve the specific version for the given subject
+func (s *Service) GetSchemaRegistrySubjectDetails(ctx context.Context, subjectName, version string) (*SchemaRegistrySubjectDetails, error) {
+	// 1. Retrieve all schema versions registered for the given subject
+	versions, err := s.getSchemaRegistrySchemaVersions(ctx, subjectName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve subject versions: %w", err)
+	}
+	latestActiveVersion := -1
+	softDeletedVersions := make(map[int]bool)
```

Comment:
minor performance nit, we can initialize capacity in this `make()` call.
```go
softDeletedVersions := make(map[int]bool, len(versions))
```

Reply:
I actually thought about it when writing the code, but I wasn't sure if it's actually a performance improvement, because the number of soft deleted versions is not known or predictable. My hypothesis is that most of the time there's no soft deleted version in a subject.